diff -urNp trafcount-7.3.orig/trafcount.c trafcount-7.3/trafcount.c
--- trafcount-7.3.orig/trafcount.c	2010-04-06 20:57:25.000000000 +0300
+++ trafcount-7.3/trafcount.c	2010-04-06 21:01:00.000000000 +0300
@@ -38,7 +38,7 @@ static void update_collector(u_int32_t i
 }
 
 static int get_counts(u_int32_t date, const char *year) {
-    struct iptc_handle *h;
+    iptc_handle_t h;
     const char *chain = NULL;
     struct collector *cur, *next;
     char db[128];
@@ -48,7 +48,7 @@ static int get_counts(u_int32_t date, co
     if (!h)
 	return 1;
 
-    for (chain = iptc_first_chain(h); chain; chain = iptc_next_chain(h)) {
+    for (chain = iptc_first_chain(&h); chain; chain = iptc_next_chain(&h)) {
 	const struct ipt_entry *e;
 	char type[IFNAMSIZ];
 	int direction;
@@ -60,7 +60,7 @@ static int get_counts(u_int32_t date, co
 	else
 	    continue;
 
-	for (e = iptc_first_rule(chain, h); e; e = iptc_next_rule(e, h)) {
+	for (e = iptc_first_rule(chain, &h); e; e = iptc_next_rule(e, &h)) {
 	    if (e->ip.src.s_addr && !e->ip.dst.s_addr && strlen(e->ip.outiface) && !strlen(e->ip.iniface)) {
 		update_collector(e->ip.src.s_addr, e->ip.outiface, e->counters.bcnt, direction);
 		snprintf(db, sizeof(db), "%s/%s/%u.%u.%u.%u.type", DBDIR, year, IP_PARTS(e->ip.src.s_addr));
@@ -73,7 +73,7 @@ static int get_counts(u_int32_t date, co
 	}
     }
 
-    iptc_free(h);
+    iptc_free(&h);
 
     for (cur = head; cur; cur = next) {
 	next = cur->next;
